migrations/20231028141843-expense-tag-stats-materialized-view.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.query(`
+      CREATE MATERIALIZED VIEW IF NOT EXISTS "ExpenseTagStats" AS
+      SELECT "tag", "count", "HostCollectiveId", null::integer AS "CollectiveId"
+      FROM (
+          SELECT unnest("Expenses".tags) AS "tag", COUNT(*) AS "count", "Collectives"."HostCollectiveId"
+          FROM "Expenses"
+          JOIN "Collectives" ON "Expenses"."CollectiveId" = "Collectives"."id"
+              AND "Collectives"."approvedAt" IS NOT NULL
+          WHERE "Collectives"."HostCollectiveId" IS NOT NULL
+          AND "Expenses"."deletedAt" IS NULL
+          AND "Expenses"."status" NOT IN ('SPAM', 'DRAFT', 'UNVERIFIED')
+          GROUP BY "tag", "Collectives"."HostCollectiveId"
+      ) AS HostTags
+      UNION 
+      SELECT "tag", "count", null::integer AS "HostCollectiveId", "CollectiveId"
+      FROM (
+          SELECT unnest(tags) AS "tag", COUNT(*) AS "count", "CollectiveId"
+          FROM "Expenses"
+          WHERE "CollectiveId" IS NOT NULL
+          AND "deletedAt" IS NULL
+          AND "status" NOT IN ('SPAM', 'DRAFT', 'UNVERIFIED')
+          GROUP BY "tag", "CollectiveId"
+      ) AS CollectiveTags;`);
+
+    await queryInterface.sequelize.query(
+      `CREATE INDEX IF NOT EXISTS expense_tag_stats__HostCollectiveId_CollectiveId ON "ExpenseTagStats"("HostCollectiveId","CollectiveId");`,
+    );
```

Comment:
Should we also create one index for the "HostCollectiveId" only query?

Reply:
The query using the table was always using both "HostCollectiveId" and "CollectiveId" (setting either to null) in the WHERE clause, but I updated the query now to only use either argument since both are not really needed, so I've added a separate index instead of this composite index for both "HostCollectiveId" and "CollectiveId".

Also added a unique composite index on "HostCollectiveId", "CollectiveId", and "tag" to make the view able to refresh CONCURRENTLY.

Reply:
Nice work!